grails-app/services/org/pih/warehouse/putaway/PutawayService.groovy

```diff
@@ -220,7 +220,7 @@ class PutawayService {
             }
         }
 
-        order.save(failOnError: true)
+        order.save(failOnError: true, flush: true)
         return order
```

Comment:
can you try just to return the line that saves? so to do:
```groovy
return order.save(failOnError: true)
```
calling save and then below returning the `order` (the not persisted one) feels weird to me
Try to do it without flush and let me know if it is working, as this flush should be unnecessary here

Reply:
Unfortunately, it does not solve the problem.
the newly added split item is still not saved at that point and so when we return the putaway after updating, it does not contain the newly added split item.

Reply:
What should help is removing @Transactional from the controller. Opening the transaction at controller level makes the transaction being commited not at the service level (so your changes made in service are not passed to the controller), but it is commited after you call `render` (so you pass not yet flushed data to the frontend). 
What should be also visible is that you should see your changes "with a delay" - what I mean by that, is that when you try to do adding the `splitItems` twice, at the second attempt, you would suddenly see the changes from the first attempt.